### PR TITLE
Implement onClick for the add/remove model buttons in the model editor

### PR DESCRIPTION
**Pull request by robertbrignull**
Hooks up the actual `onClick` handler for the add/remove model buttons.

Also adds tests. I couldn't think of any more edge cases to test, but if there are some please let me know and I'll add tests for those.

(While testing, I did spot one bug where the method modeling panel didn't update perfectly. I'll open a separate PR for this.)

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by koesie10 — COMMENTED** [review]
I think the tests will need to be changed since re-rendering won't happen automatically, but the code itself looks good.

File: extensions/ql-vscode/src/view/model-editor/__tests__/MethodRow.spec.tsx [hunk 1]
```diff
@@ -358,4 +358,67 @@ describe(MethodRow.name, () => {
...
+      modeledMethods: [modeledMethod],
+      viewState: {
+        ...viewState,
+        showMultipleModels: true,
+      },
+    });
+
+    await userEvent.click(screen.getByLabelText("Add new model"));
+
+    const kindInputs = screen.getAllByRole("combobox", { name: "Model type" });
```
[the remark this answers is not in the document]
Comment (robertbrignull) [reply extensions/ql-vscode/src/view/model-editor/__tests__/MethodRow.spec.tsx:373]: Thanks. I also came to the same conclusion. I've just got all the tests working using the `onChange` handler.

File: extensions/ql-vscode/src/view/model-editor/MethodRow.tsx [hunk 2]
```diff
@@ -126,6 +126,35 @@ const ModelableMethodRow = forwardRef<HTMLElement | undefined, MethodRowProps>(
...
 
+    const removeModelClickedHandlers = useMemo(
+      () =>
+        modeledMethods
+          .slice(0, modeledMethods.length - 1)
+          .map((_, index) => () => {
+            const newModeledMethods = [...modeledMethods];
+            newModeledMethods.splice(index, 1);
+            onChange(method.signature, newModeledMethods);
+          }),
```
[the remark this answers is not in the document]
Comment (robertbrignull) [reply extensions/ql-vscode/src/view/model-editor/MethodRow.tsx:137]: Sure, that's fine. Technically we don't need that final handler, but I agree the code is a little cleaner if we just create it anyway. I'm fine with that.

**Review by koesie10 — APPROVED** [review]
